promql/testdata/functions.test

```diff
@@ -71,24 +71,63 @@ load 5m
 	testcounter_reset_middle	0+10x4 0+10x5
 	http_requests{job="app-server", instance="1", group="canary"}		0+80x10
 
-# Deriv should return the same as rate in simple cases.
+# deriv should return the same as rate in simple cases.
 eval instant at 50m rate(http_requests{group="canary", instance="1", job="app-server"}[60m])
 	{group="canary", instance="1", job="app-server"} 0.26666666666666666
 
 eval instant at 50m deriv(http_requests{group="canary", instance="1", job="app-server"}[60m])
 	{group="canary", instance="1", job="app-server"} 0.26666666666666666
 
-# Deriv should return correct result.
+# deriv should return correct result.
 eval instant at 50m deriv(testcounter_reset_middle[100m])
 	{} 0.010606060606060607
 
-# Predict_linear should return correct result.
+# predict_linear should return correct result.
 eval instant at 50m predict_linear(testcounter_reset_middle[100m], 3600)
 	{} 88.181818181818185200
 
-# Predict_linear is syntactic sugar around deriv.
+# predict_linear is syntactic sugar around deriv.
 eval instant at 50m predict_linear(http_requests[50m], 3600) - (http_requests + deriv(http_requests[50m]) * 3600)
 	{group="canary", instance="1", job="app-server"} 0
 
 eval instant at 50m predict_linear(testcounter_reset_middle[100m], 3600) - (testcounter_reset_middle + deriv(testcounter_reset_middle[100m]) * 3600)
 	{} 0
+
+clear
+
+# Tests for label_replace.
+load 5m
+  testmetric{src="source-value-10",dst="original-destination-value"} 0
+  testmetric{src="source-value-20",dst="original-destination-value"} 1
+
+# label_replace does a substring match and replace.
+eval instant at 0m label_replace(testmetric, "dst", "destination-value-$1", "src", "value-(.*)")
+  testmetric{src="source-value-10",dst="destination-value-10"} 0
+  testmetric{src="source-value-20",dst="destination-value-20"} 1
+
+# label_replace works with multiple capture groups.
+eval instant at 0m label_replace(testmetric, "dst", "$1-value-$2", "src", "(.*)-value-(.*)")
+  testmetric{src="source-value-10",dst="source-value-10"} 0
+  testmetric{src="source-value-20",dst="source-value-20"} 1
+
+# label_replace does not overwrite the destination label if the source label
```

Comment:
It should work if the source label is empty, and the regex matches it. Can you add a test for this?